### ISSUE TITLE
generate error images when inference

### ISSUE THREAD

**Issue by Blackkinggg** [RenYurui/Neural-Texture-Extraction-Distribution#8]
I can generated right images using the demo.py, but get below images when used the inference.py
![1653305889(1)](https://user-images.githubusercontent.com/39241441/169811216-4fcfa532-9892-4891-9730-e9f2fc36f995.jpg)

**Comment by Blackkinggg** [comment]
The same problem also appears in the training stage, it seems that attention list is empty.
![image](https://user-images.githubusercontent.com/39241441/169970580-2eed301e-abae-4553-8fa5-d4042fe16734.png)

**Comment by Blackkinggg** [comment]
There are also problems in the training stage

### 
def attn2image(source_softmax, seg_query, input_image):
    target_list = []
    image_size = input_image.shape[2:]
    for softmax, query in zip(source_softmax, seg_query):
        b, num_label, h, w = query.shape
        input_resize = F.interpolate( input_image, (h,w) )
        input_resize = input_resize.view(b, -1, h*w)
        extracted = torch.einsum('bkm,bvm->bvk', softmax, input_resize)
        query = F.softmax(query.view(b, num_label, -1), 1)
        estimated_target = torch.einsum('bkm,bvk->bvm', query, extracted)
        estimated_target = estimated_target.view(b, -1, h, w)
        target_list.append(F.interpolate(estimated_target, image_size))
    target_gen = torch.cat(target_list, 3)
    return target_gen 
### 
The source_softmax is an empty list.

**Comment by RenYurui** [comment]
Hi!
Is the checkpoint loaded successfully while inference?
Can you provide the messages when you run inference.py

**Comment by Blackkinggg** [comment]
> Hi! Is the checkpoint loaded successfully while inference? Can you provide the messages when you run inference.py

您好，我使用的原始代码，看起来inference没有成功load模型，没有报错信息，直接生成上面的图像

**Comment by RenYurui** [comment]
When you run `inference.py` without training the model. You may need to load the model by adding `--which_iter` such as:

``` bash
python -m torch.distributed.launch \
--nproc_per_node=1 \
--master_port 12345 inference.py \
--config ./config/fashion_512.yaml \
--name fashion_512 \
--no_resume \
--output_dir ./result/fashion_512/inference \
--which_iter 495400
```

**Comment by SHULJJ** [comment]
hello ，请问你解决了这个问题吗
![image](https://user-images.githubusercontent.com/86544674/199534974-d0d5d39d-8060-4088-8f92-780bb3bb0e2b.png)

**Comment by SHULJJ** [comment]
> There are also problems in the training stage. Neural-Texture-Extraction-Distribution/util/visualization/linear_attention.py:
> 
> def attn2image(source_softmax, seg_query, input_image): target_list = [] image_size = input_image.shape[2:] for softmax, query in zip(source_softmax, seg_query): b, num_label, h, w = query.shape input_resize = F.interpolate( input_image, (h,w) ) input_resize = input_resize.view(b, -1, h*w) extracted = torch.einsum('bkm,bvm->bvk', softmax, input_resize) query = F.softmax(query.view(b, num_label, -1), 1) estimated_target = torch.einsum('bkm,bvk->bvm', query, extracted) estimated_target = estimated_target.view(b, -1, h, w) target_list.append(F.interpolate(estimated_target, image_size)) target_gen = torch.cat(target_list, 3) return target_gen
> 
> The source_softmax is an empty list.
请问你解决这个问题了吗

**Comment by shrinivasait** [comment]
Hi @Blackkinggg, How did generate pose key points?
